src/structures/AutocompleteInteraction.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+const CommandInteractionOptionResolver = require('./CommandInteractionOptionResolver');
+const Interaction = require('./Interaction');
+
+/**
+ * Represents a autocomplete interaction.
+ * @extends {Interaction}
+ */
+class AutocompleteInteraction extends Interaction {
```

Comment:
Is this tested? `transformOption` and `transformResolved` don't exist on Interaction.

Reply:
It might be able to extend `CommandInteraction` based on the params

Reply:
Oh, I don't think the best option is to extend `BaseCommandInteraction` though, there's too many properties that don't apply to this interaction type. Need to find some way to have the relevant properties and methods without too much code duplication.

Reply:
Update, it appears resolved doesn't exist for these interactions, so the code duplication could be minimal. I think that would be the best approach.